doc/tutorial/concepts/index.md

```diff
@@ -0,0 +1,214 @@
+# C++ Concepts {#tutorial_concepts}
+
+[TOC]
+
+This tutorial introduces "C++ Concepts", a feature of C++20 (and available to some extent in older GCC versions).
+You will learn the terminology used in the context of concepts and how to use SeqAn3's concepts in your application.
+
+\tutorial_head{Moderate, 60 min, \ref setup\, \ref tutorial_alphabets,}
+
+This tutorial teaches the very basics of working with concepts. For more background and information on how to implement
+your own concepts, we recommend:
+  * A well-readable [paper](http://www.stroustrup.com/good_concepts.pdf) with motivation and historical background.
+  * The (rather dense) [documentation on cppreference](https://en.cppreference.com/w/cpp/language/constraints).
+
+
+# Constraints
+
+## Introduction
+
+In C++20 a typical pre-C++20 template is said to be *unconstrained*, i.e. it can be instantiated with any type:
+
+```cpp
+template <typename t>
+t add(t const v1, t const v2)
+{
+    return v1 + v2;
+}
+
+int main()
+{
+    return add(1, 3); // instantiates add<int>()
+}
+```
+
+Of course this will not actually compile for all types, because the function template **implicitly requires** that
+types provide a `+` operator.
+*Constraints* are a way of making requirements of template arguments **explicit**.
+Constraints can be formulated ad-hoc, but this tutorial only covers *concepts*.
+Concepts are a set of constraints with a given name.
+Let's assume there is a concept called `Addable` that requires the existance of a `+` operator (as previously mentioned
+the syntax for defining concepts is not covered here).
+The following snippet demonstrates how we can constrain our function template, i.e. make the template immediately
+reject any types that don't satisfy the requirement:
+
+```cpp
+template <Addable t>
+t add(t const v1, t const v2)
+{
+    return v1 + v2;
+}
+
+int main()
+{
+    return add(1, 3); // instantiates add<int>()
+}
+```
+
+The only difference is that we have replaced `typename` with `Addable`. The standard library provides a set of
+[predefined concepts](https://en.cppreference.com/w/cpp/concepts), for our example above, the std::Integral concept
+could have been used.
+
```

Comment:
Maybe add. 
```
You can test if a type models a concept by using `static_assert(Concept<type>)`

snippet example with integral.
```

Reply:
This is explained futher down on the page.